api/queues.js

```diff
@@ -94,6 +93,28 @@ router.get(
   }
 )
 
+// Modify a queue for a course
+router.patch(
+  '/:queueId',
+  [
+    requireCourseStaffForQueue,
+    requireQueue,
+    check('name').isLength({ min: 1 }),
+    check('location').optional({ nullable: true }),
```

Comment:
This should probably take into account the fact that queues can be fixed-location now, in which case we want to mandate that a location be set.

Reply:
Take a look at https://github.com/illinois/queue/blob/master/api/questions.js#L29, you'd want to check if the queue is fixed-location in a custom validator and only require it if it's fixed-location.

Reply:
Oh, I see how it works, will do that after dinner. Thanks!

Reply:
Done